packages/wallet/backend/src/account/service.ts

```diff
@@ -132,14 +129,18 @@ export class AccountService implements IAccountService {
       user.rapydWalletId
     )
 
-    accounts.forEach((acc) => {
-      acc.balance = transformBalance(
+    for (let i = 0; i < accounts.length; i++) {
+      accounts[i].balance = transformBalance(
         accountsBalance.data?.find(
-          (rapydAccount) => rapydAccount.currency === acc.assetCode
+          (rapydAccount) => rapydAccount.currency === accounts[i].assetCode
         )?.balance ?? 0,
-        acc.assetScale
+        accounts[i].assetScale
       )
-    })
+      accounts[i].paymentPointers = await PaymentPointer.query().where(
```

Comment:
Just another option here, that would have bettern performance but worse maintainability IMO:

make a single initial query that gets all payment pointers of all accounts of a user (since we have the userId) -> one to many with accounts -> one to many with payment pointers. (a couple of sql joins needed)

This way you don't get the tcp overhead of every separate sql query, and also benefit on the db engine optimisations.

However, since we won't be running in any performance issues with the scope of our project, both options are valid for me.
@raducristianpopa  @dragosp1011